src/cbor.zig

```diff
@@ -293,6 +326,36 @@ pub const MapIterator = struct {
     }
 };
 
+pub const IndefMapIterator = struct {
+    data: []const u8,
+    i: usize,
+
+    /// Get the next key Pair
+    ///
+    /// Returns null after the last element.
+    pub fn next(self: *@This()) ?Pair {
+
```

Comment:
I'd add `if (self.i >= self.data.len) return null;` just in case. While Zig catches overreads via panics, this guard helps us handling out of bound errors more gracefully.